src/EnergyPlus/InternalHeatGains.cc

```diff
@@ -7762,8 +7764,17 @@ namespace InternalHeatGains {
                     WAirIn = state.dataHeatBalFanSys->ZoneAirHumRat(NZ);
                 } else {
                     // TAirIn = TRoomAirNodeIn, according to EngineeringRef 17.1.4
-                    int ZoneAirInletNode = state.dataZoneEquip->ZoneEquipConfig(NZ).InletNode(1);
-                    TSupply = state.dataLoopNodes->Node(ZoneAirInletNode).Temp;
+                    if (!state.dataHeatBal->ZoneITEq(Loop).gotControlledZoneIndex) { // Do the following only once to save time
+                        state.dataHeatBal->ZoneITEq(Loop).zoneEqIndex =
+                            DataZoneEquipment::GetControlledZoneIndex(state, state.dataHeatBal->Zone(state.dataHeatBal->ZoneITEq(Loop).ZonePtr).Name);
+                        state.dataHeatBal->ZoneITEq(Loop).gotControlledZoneIndex = true;
+                    }
```

Comment:
I was thinking of doing this around line 2918 where the zoneEqIndex is already found early on. Initializing there would remove this conditional here at lines 7767-7771 and also remove the need for the new bool gotControlledZoneIndex. Unless there is some reason zoneEqIndex cannot be reliably found there, of course.

Reply:
Well, that is definitely what I should have done before--much cleaner and no need for the extra bool variable.  Kicking myself because I had looked at that code at one point but then forgot that it was there.  The code around line 2918 basically is assuming the information is available.  If not, there's another problem with the code.  Fixed this and made another commit just now.  Hopefully CI comes back clean.